Explorer/Assets/Scripts/Global/Dynamic/RealmNavigator.cs

```diff
@@ -82,107 +94,188 @@ public async UniTask<bool> TryChangeRealmAsync(URLDomain realm, CancellationToke
+                    else
+                        await genesisTerrain.ShowAsync(landscapeLoadReport);
+                }
+                else
+                {
+                    genesisTerrain.Hide();
+                    await GenerateWorldTerrainAsync((uint)realmController.GetRealm().GetHashCode(), landscapeLoadReport, ct);
+                }
+            }
+        }
 
-            SwitchMiscVisibilityAsync(isGenesis);
+        private async UniTask<UniTask> TeleportToParcelAsync(Vector2Int parcel, AsyncLoadProcessReport processReport, CancellationToken ct)
+        {
+            WaitForSceneReadiness? waitForSceneReadiness = await teleportController.TeleportToSceneSpawnPointAsync(parcel, processReport, ct);
+            return waitForSceneReadiness.ToUniTask(ct);
+        }
 
-            var postRealmLoadReport = AsyncLoadProcessReport.Create();
+        private async UniTask<UniTask> TeleportToWorldSpawnPointAsync(AsyncLoadProcessReport processReport, CancellationToken ct)
+        {
+            AssetPromise<SceneEntityDefinition, GetSceneDefinition>[] promises = await WaitForFixedScenePromisesAsync(ct);
+            WaitForSceneReadiness? waitForSceneReadiness = await teleportController.TeleportToSceneSpawnPointAsync(promises[0].Result!.Value.Asset!.metadata.scene.DecodedBase, processReport, ct);
+            return waitForSceneReadiness.ToUniTask(ct);
+        }
 
-            await UniTask.WhenAll(postRealmLoadReport.PropagateAsync(loadReport, ct, loadReport.ProgressCounter.Value, timeout: TimeSpan.FromSeconds(30)),
-                isGenesis
-                    ? genesisTerrain.ShowAsync(postRealmLoadReport)
-                    : GenerateWorldTerrainAsync((uint)realm.GetHashCode(), postRealmLoadReport, ct));
+        private async UniTask ChangeRealmAsync(URLDomain realm, CancellationToken ct)
+        {
+            await realmController.SetRealmAsync(realm, ct);
+            await SwitchMiscVisibilityAsync();
         }
 
         private async UniTask GenerateWorldTerrainAsync(uint worldSeed, AsyncLoadProcessReport processReport, CancellationToken ct)
         {
             if (!worldsTerrain.IsInitialized)
                 return;
 
-            await UniTask.WaitUntil(() => realmController.GlobalWorld.EcsWorld.Get<FixedScenePointers>(realmController.RealmEntity).AllPromisesResolved, cancellationToken: ct);
-
-            AssetPromise<SceneEntityDefinition, GetSceneDefinition>[] promises = realmController.GlobalWorld.EcsWorld.Get<FixedScenePointers>(realmController.RealmEntity).Promises;
+            AssetPromise<SceneEntityDefinition, GetSceneDefinition>[] promises = await WaitForFixedScenePromisesAsync(ct);
 
             var decodedParcelsAmount = 0;
 
             foreach (AssetPromise<SceneEntityDefinition, GetSceneDefinition> promise in promises)
+            {
+                //TODO (JUANI) : Shouldnt be possible, but some promises result are null
```

Comment:
Can we investigate it? I may cause soem other issues

Reply:
I wasnt able to reproduce it again, Im able to teleport back and forth from goerli with no problem. Probably I was missing an await when developing. I ll remove this code

Reply:
I tried to reproduce it but had no success either